playground/src/main.tsx

```diff
@@ -53,6 +56,9 @@ function App(props: { katas: Kata[]; linkedCode?: string }) {
   );
   const [evtTarget] = useState(new QscEventTarget(true));
 
+  const languageService = getLanguageServiceWorker(languageServiceWorkerPath);
```

Comment:
As it's a React component, this render function re-runs whenever props or state change. As such, I think you'll find this line and the next probably run every time state changes (e.g. when the compiler changes from 'idle' to 'busy' or back as one example).

Being the language service never changes currently, you likely want to do lines 59 and 60 in the `loaded` function, and pass in the language service as a propery, or follow the pattern used to create the compiler on line 54. (The `useState` initializer only runs on first render).

Reply:
I see. Thanks. I totally missed the point of the `useState` pattern above.

Reply:
Incidentally, what about line 57 above? Is it intentional that `new QscEventTarget(true))` runs on every state change? I think `useState` only uses that as the initial value, so it's harmless, but perhaps it should be wrapped in a callback?

Reply:
Good catch. That's probably a bug. Feel free to fix here (and verify), else I can do that later.